nexus/internal/uploader/uploader_default.go

```diff
@@ -42,24 +45,46 @@ func (u *DefaultUploader) Upload(task *UploadTask) error {
 		}
 	}(file)
 
-	req, err := retryablehttp.NewRequest(
-		http.MethodPut,
-		task.Url,
-		file,
-	)
-
+	fileWithLen, err := NewFileWithLen(file)
+	if err != nil {
+		return err
+	}
+	req, err := retryablehttp.NewRequest(http.MethodPut, task.Url, fileWithLen)
+	if err != nil {
+		return err
+	}
 	for _, header := range task.Headers {
 		parts := strings.Split(header, ":")
 		req.Header.Set(parts[0], parts[1])
 	}
 
+	resp, err := u.client.Do(req)
 	if err != nil {
 		return err
 	}
-
-	if _, err = u.client.Do(req); err != nil {
-		return err
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		return fmt.Errorf("response: %v", resp.Status)
```

Comment:
why do you need this? we already have these error codes handled in the client

Reply:
Without this change the server returns HTTP 411 error, but we never check it and just return `nil`.
Did you expect `u.client.Do(req)` above to return an error in this case? That's not what is happening.

Reply:
ok, sorry we only implemented it for the graphql client. 
could you do something similar to what we did for the graphql client: `clients.WithRetryClientRetryPolicy(clients.CheckRetry)` see this code path

Reply:
Hm, I don't think this is what we should be doing. The retry policy is supposed to decide whether to retry the request. Yes, any errors encountered during this check are returned from `client.Do`. This doesn't mean the retry policy should return an error when the request failed. Note that the default policy does not do it.

Since we already did this elsewhere, I'll follow the same pattern here. I think it should be changed though.